src/Form/PhoneInput.js

```diff
@@ -0,0 +1,61 @@
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { AsYouType } from 'libphonenumber-js';
+import Input from './Input';
+import { createEasyInput } from './EasyInput';
+import { getNextCursorPosition, isDeletingCharacter } from '../utils';
+
+const formatPhoneNumber = (countryCode, value = '') => {
+  const formatter = new AsYouType(countryCode);
+  return formatter.input(value.replace(/\D/g, ''));
+};
+
+function PhoneInput({ forwardedRef, value: propValue, onChange, countryCode, ...inputProps }) {
+  const format = (newValue, oldValue) => formatPhoneNumber(countryCode, newValue, oldValue);
+  const [currentValue, setValue] = useState(format(propValue));
+  const ref = forwardedRef || useRef();
+  const cursorPosition = useRef(currentValue.length);
+
+  useEffect(() => {
+    if (propValue !== currentValue) {
+      setValue(format(propValue, currentValue));
+    }
+  }, [propValue]);
+
+  useEffect(() => {
+    if (ref.current) {
+      ref.current.setSelectionRange(cursorPosition.current, cursorPosition.current);
+    }
+  }, [currentValue]);
+
+  const handleChange = (name, newValue, event) => {
+    const isDeletingParen = isDeletingCharacter(
+      ')',
+      newValue,
+      currentValue,
+      event.target.selectionEnd || currentValue.length
+    );
+    const formattedValue = isDeletingParen ? newValue : format(newValue, currentValue);
+
+    cursorPosition.current = getNextCursorPosition(event.target.selectionEnd, formattedValue, currentValue);
+
+    setValue(formattedValue);
+
+    if (onChange) {
+      onChange(name, formattedValue);
+    }
+  };
+
+  return <Input forwardedRef={ref} value={currentValue} onChange={handleChange} {...inputProps} />;
```

Comment:
1. Should we set a `maxLength` of 10 numbers (or 11 with country code)?
2. Can we make this input `type="tel"` by default so we can have a numpad on mobile devices?

Reply:
Should we specify the expected format as a parenthetical aspect in the label (e.g. (123) 123-1234). Should we also allow the character required for the format? For example, if a user types in "(" they are currently blocked from doing so, which might be unintuitive if the format is specified in the label. Or maybe not. Perhaps we could consider masking (https://www.uxbooth.com/articles/the-new-rules-of-form-design/, https://uxplanet.org/designing-perfect-text-field-clarity-accessibility-and-user-effort-d03c1e26004b).

Reply:
That behavior is dictated by Google's `libphonenumber` to disallow the leading paren so I'd rather not get in the way. It's also more difficult to type a paren on mobile when the numpad is presented.